pkg/route/secretmanager/manager.go

```diff
@@ -0,0 +1,148 @@
+package secretmanager
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/openshift/library-go/pkg/route/secret"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+type SecretManager interface {
```

Comment:
I've refactored `secretmanager` a bit to add an interface and also to add a `fake` for it. Fakes might be required by its consumer.

xRef : https://github.com/openshift/library-go/pull/1626/commits/a417674c400d5866bd59fdfad0f84458b53c8958